crates/biome_js_analyze/src/lint/nursery/use_explicit_type.rs

```diff
@@ -275,11 +314,60 @@ declare_node_union! {
         | TsGetterSignatureClassMember
         | TsDeclareFunctionDeclaration
         | TsDeclareFunctionExportDefaultDeclaration
+        | JsFormalParameter
+        | JsRestParameter
+        | TsThisParameter
+}
+
+pub enum UseExplicitTypeState {
+    MissingReturnType(TextRange),
+    MissingArgumentnType(TextRange, String),
```

Comment:
to continue the discussion from [this](https://github.com/biomejs/biome/pull/4463#discussion_r1831008161)

@Conaclos 

Thank you for the feedback! I decided to keep the current implementation for the following reasons:

- Even if we use the token directly instead of passing a String, we still need to call the to_string() method to pass the value to markup! {}, which will allocate a string regardless. So, both approaches result in string allocation.
- Additionally, in cases like function foo({a: b}), we need to handle tokens differently. For instance, the first token in this scenario would be '{'.